### PR TITLE
Creating new method to report removed vulnerabilities

### PR DESCRIPTION
**Pull request by pereyra-m**
|Related issue|
|---|
|#7925|

## Description

This PR adds a new method to report the removed vulnerabilities of the agents.

## Logs/Alerts example

Here is an example of the alert generated
```
** Alert 1617854022.1363930: - vulnerability-detector,gdpr_IV_35.7.d,pci_dss_11.2.1,pci_dss_11.2.3,tsc_CC7.1,tsc_CC7.2,
2021 Apr 08 03:53:42 (local_centos) 127.0.0.1->vulnerability-detector
Rule: 23502 (level 3) -> 'The CVE-2019-17594 that affected ncurses was eliminated due to a package removal/update or a system upgrade'
{"vulnerability":{"package":{"name":"ncurses","version":"5.9-14.20130511.el7_4","architecture":"x86_64"},"cve":"CVE-2019-17594","status":"Removed","reference":"1234","type":"UNDEFINED","title":"CVE-2019-17594 affecting ncurses was eliminated"}}
vulnerability.package.name: ncurses
vulnerability.package.version: 5.9-14.20130511.el7_4
vulnerability.package.architecture: x86_64
vulnerability.cve: CVE-2019-17594
vulnerability.status: Removed
vulnerability.reference: 1234
vulnerability.type: UNDEFINED
vulnerability.title: CVE-2019-17594 affecting ncurses was eliminated
```

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux
 - [x] Source installation

- Memory tests for Linux
  - [x] Scan-build report
  - [x] Valgrind (memcheck and descriptor leaks check)
- [x] Added unit tests (for new features)

### CODE REVIEW

**Comment by pereyra-m** [comment]
### Tests

#### Valgrind

The analysis with this tool was successful
valgrind_sucess

#### Scan-build

The tool didn't report any error
scan_build_sucess

File: src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c [hunk 1]
```diff
@@ -1567,6 +1570,72 @@ int wm_vuldet_send_cve_report(vu_report *report) {
...
+    cJSON_AddItemToObject(j_alert, "vulnerability", j_alert_cve);
+    cJSON_AddItemToObject(j_alert_cve, "package", j_package);
+
+    if (j_alert && j_alert_cve && j_package) {
+        cJSON_AddStringToObject(j_package, "name", package_name);
+        cJSON_AddStringToObject(j_package, "version", package_version);
+        cJSON_AddStringToObject(j_package, "architecture", package_architecture);
+        cJSON_AddStringToObject(j_alert_cve, "cve", cve_id);
+        // To trigger the alert, the status is defined here
+        cJSON_AddStringToObject(j_alert_cve, "status", "Removed");
```
Comment (DProvinciani) [inline src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c:1604]: Although this is the only one place where the `REMOVED` status is used, I would define it in the `wdb.h` header file together with the other statuses and use the definition here.

```c
#define VULN_CVES_STATUS_REMOVED "REMOVED"
```

Comment (DProvinciani) [reply src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c:1604]: Probably this will require changing the string to capital letters in the rule.

**Review by DProvinciani — CHANGES_REQUESTED** [review]
Please, apply changes according to the comments.

File: src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c (hunk 1 above)
Comment (palaciosjeremias) [reply src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c:1604]: +1

**Review by palaciosjeremias — APPROVED** [review]

File: src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c (hunk 1 above)
Comment (pereyra-m) [reply src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c:1604]: You are right, it is much better with a definition.
Fixed !

**Pull request merged**